stdlib/public/core/StringRangeReplaceableCollection.swift.gyb

```diff
@@ -44,7 +44,7 @@ extension String : StringProtocol, RangeReplaceableCollection {
   
   // This initializer satisfies the LosslessStringConvertible conformance
   @available(swift, obsoleted: 4, message: "String.init(_:String) is no longer failable")
-  public init?(_ other: String) {
+  public init?(_ other: String, obsoletedInSwift4: () = ()) {
```

Comment:
This hack prevents the `init?` from being used to satisfy a protocol requirement; you can satisfy a protocol requirement with something that's not yet available, but not with something that has become unavailable.  IOW, without this, we can't throw the switch on the Swift 4 transition.  I should add a comment to make that clear.